projects/smoke_thermistor/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// Toggle the following to select which data is logged
+#define READ_DATA_RAW true
+#define READ_DATA_CONVERTED false
+#define NUM_ADC_PINS 6
+#define TIME_BETWEEN_READS_IN_MILLIS 200
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+
+  GpioAddress pin_addresses[NUM_ADC_PINS];
+
+  for (uint8_t pin = 0; pin < NUM_ADC_PINS; pin++) {
+    pin_addresses[pin].port = GPIO_PORT_A;
+    pin_addresses[pin].pin = pin;
+  }
```

Comment:
IIRC the thermistor pins are on A0-A5 on solar (though A365 is down for maintenance so I can't check, lol), but if we ever want to use this smoke test for other pins, this kind of an opaque way of initializing the addresses and it's hard to change the pins we're reading from. Instead I'd prefer `pin_addresses` to be a static array with an explicit list of the `GpioAddress`es to read from (i.e. `GpioAddress s_pin_addresses[] = { { GPIO_PORT_A, 0}, ... };`) so that it's easier to change.